src/Service/QuerySearchService.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\CoreBundle\Service;
+
+use EMS\CommonBundle\Helper\Text\Encoder;
+use EMS\CoreBundle\Entity\QuerySearch;
+use EMS\CoreBundle\Repository\QuerySearchRepository;
+use Psr\Log\LoggerInterface;
+
+final class QuerySearchService implements EntityServiceInterface
+{
+    private QuerySearchRepository $querySearchRepository;
+    private LoggerInterface $logger;
+
+    public function __construct(QuerySearchRepository $querySearchRepository, LoggerInterface $logger)
+    {
+        $this->querySearchRepository = $querySearchRepository;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @return QuerySearch[]
+     */
+    public function getAll(): array
+    {
+        return $this->querySearchRepository->getAll();
+    }
+
+    public function update(QuerySearch $querySearch): void
+    {
+        if (0 === $querySearch->getOrderKey()) {
+            $querySearch->setOrderKey($this->querySearchRepository->counter() + 1);
+        }
+        $encoder = new Encoder();
+        $name = $querySearch->getName();
+        if (null === $name) {
+            throw new \RuntimeException('Unexpected null name');
+        }
+        $webalized = $encoder->webalize($name);
+        if (null === $webalized) {
+            throw new \RuntimeException('Unexpected null webalized name');
+        }
+        $querySearch->setName($webalized);
+        $this->querySearchRepository->create($querySearch);
+    }
+
+    public function delete(QuerySearch $querySearch): void
+    {
+        $name = $querySearch->getName();
+        $this->querySearchRepository->delete($querySearch);
+        $this->logger->warning('log.service.query_search.delete', [
+            'name' => $name,
+        ]);
+    }
+
+    /**
+     * @param string[] $ids
+     */
+    public function deleteByIds(array $ids): void
+    {
+        foreach ($this->querySearchRepository->getByIds($ids) as $querySearch) {
+            $this->delete($querySearch);
+        }
+    }
+
+    /**
+     * @param array<string, int> $ids
+     */
+    public function reorderByIds(array $ids): void
+    {
+        foreach ($this->querySearchRepository->getByIds(\array_keys($ids)) as $querySearch) {
+            $querySearch->setOrderKey($ids[$querySearch->getId()] ?? 0);
```

Comment:
The first item has the index 0, but on 34 if the order field is equal to 0 the order is set to the last:
```suggestion
            $querySearch->setOrderKey(isset($ids[$querySearch->getId()]) ? $ids[$querySearch->getId()] + 1  : 0);
```